src/commands/utils/metar.ts

```diff
@@ -15,61 +16,49 @@ export const metar: CommandDefinition = {
 
-            if (response.statusCode === 200) {
-                // Response OK, parse the JSON
-                const metarReport = JSON.parse(body);
-                metarEmbed = makeEmbed({
-                    title: `METAR Report | ${metarReport.station}`,
-                    description: makeLines([
-                        '**Raw Report**',
-                        metarReport.raw,
-                        '',
-                        '**Basic Report:**',
-                        `**Time Observed:** ${metarReport.time.dt}`,
-                        `**Station:** ${metarReport.station}`,
-                        `**Wind:** ${metarReport.wind_direction.repr}${metarReport.wind_direction.repr === 'VRB' ? '' : Units.DEGREES} at ${metarReport.wind_speed.repr} ${Units.KNOTS}`,
-                        `**Visibility:** ${metarReport.visibility.repr} ${Number.isNaN(+metarReport.visibility.repr) ? '' : metarReport.units.visibility}`,
-                        `**Temperature:** ${metarReport.temperature.repr} ${Units.CELSIUS}`,
-                        `**Dew Point:** ${metarReport.dewpoint.repr} ${Units.CELSIUS}`,
-                        `**Altimeter:** ${metarReport.altimeter.value.toString()} ${metarReport.units.altimeter}`,
-                        `**Flight Rules:** ${metarReport.flight_rules}`,
-                    ]),
-                    fields: [
-                        {
-                            name: 'Unsure of how to read the raw report?',
-                            value: 'Please refer to our guide [here.](https://docs.flybywiresim.com/pilots-corner/airliner-flying-guide/weather/)',
-                            inline: false,
-                        },
-                    ],
-                    footer: { text: 'This METAR report may not accurately reflect the weather in the simulator. However, it will always be similar to the current conditions present in the sim.' },
-                });
-            } else if (response.statusCode === 400) {
-                // Invalid ICAO/IATA code
-                metarEmbed = makeEmbed({
-                    title: `METAR Error | ${icaoArg.toUpperCase()}`,
-                    description: makeLines([
-                        `${icaoArg.toUpperCase()} is not a valid station code!`,
-                    ]),
-                });
-            } else {
-                // Unknown error
-                metarEmbed = makeEmbed({
-                    title: 'METAR Error',
-                    description: makeLines([
-                        'There was an unknown error with the METAR request!',
-                    ]),
-                });
+        try {
+            const metarReport = await fetch(`https://avwx.rest/api/metar/${icaoArg}`, {
+                method: 'GET',
+                headers: { Authorization: process.env.METAR_TOKEN },
+            }).then((res) => res.json());
+
+            if (metarReport.error) {
+                await msg.reply(metarReport.error);
```

Comment:
Is it possible to have the errors displayed in an embed like they were before? Do you think it looks nicer in an embed? (Could do all errors in these commands in embeds?)

Reply:
I personally prefer embeds over replies. I'll work on making embeds for all soon